### DISCUSSION TITLE
能否提供阿里云所有docker镜像的列表以及每个docker镜像的描述？

### DISCUSSION THREAD

**Discussion by hh2712** [alibaba/GraphScope#1733] [Q&A] [answered]
1.能否在文档中增加对docker镜像的描述？
2.如何能够访问到阿里云的私有docker仓库？
3. 当前是否提供了支持pytorch的graphscope镜像的版本？

**Comment by welcome** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! And a maintainer will get back to you shortly!
Please feel free to contact us on [DingTalk](https://h5.dingtalk.com/circle/healthCheckin.html?dtaction=os&corpId=dingb4b76e0fa9ccd8d516fd0f0acdf19f55&68e71190=0fe34ec0&cbdbhh=qwertyuiop), WeChat account(graphscope) or [Slack](http://slack.graphscope.io/). We are happy to answer your questions responsively.

**Comment by lidongze0629** [comment]
@hh2712 ,  感谢你对 GraphScope 项目的关注，关于你的问题:

**1. 能否在文档中增加对docker镜像的描述?**

我们争取在下个版本增加对 GraphScope Docker 镜像的描述，这里简单介绍如下:

- `registry.cn-hongkong.aliyuncs.com/graphscope/graphscope`: 包含了 GraphScope 所有的服务端组件部分，可直接部署在 K8s 上，也是 [Helm](https://github.com/alibaba/GraphScope/tree/main/charts/graphscope) 部署时需要的镜像
- `registry.cn-hongkong.aliyuncs.com/graphscope/graphscope-store`: GraphScope 存储服务相关镜像，可由 [Helm](https://github.com/alibaba/GraphScope/tree/main/charts/graphscope-store) 部署
- `registry.cn-hongkong.aliyuncs.com/graphscope/graphscope-vineyard`: 开发镜像，具备编译 GraphScope 的所有依赖环境


**2. 如何能够访问到阿里云的私有docker仓库?**

GraphScope 所有镜像都是公共的，可直接访问，如果你想访问自己的aliyun私有镜像，可参考[官方文档](https://www.alibabacloud.com/help/zh/container-registry)

**3. 当前是否提供了支持 pytorch 的 graphscope 镜像的版本**

GraphScope 目前还有支持 pytorch 的镜像版本，若想使用 pytorch，需要重新实现已有的模型。我们也计划在未来的 GraphScope 版本中支持 pytorch

**Comment by hh2712** [comment]
感谢您的回答。
初学docker不久，请教一下registry.cn-hongkong.aliyuncs.com/graphscope应该是你们搭建的docker私有仓库吧？尝试过docker pull，可以拉取对应tag的镜像，但是暂时没有找到查询registry.cn-hongkong.aliyuncs.com/graphscope目录下所有镜像列表的方式。

**Comment by lidongze0629** [comment, marked as answer]
> 感谢您的回答。 初学docker不久，请教一下registry.cn-hongkong.aliyuncs.com/graphscope应该是你们搭建的docker私有仓库吧？尝试过docker pull，可以拉取对应tag的镜像，但是暂时没有找到查询registry.cn-hongkong.aliyuncs.com/graphscope目录下所有镜像列表的方式。

是的，目前还无法查询 `registry.cn-hongkong.aliyuncs.com/graphscope` 下的所有 `tag`，我们计划会在 [dockerhub](https://hub.docker.com/) 上同步一份，现阶段每个 [GraphScope Release](https://github.com/alibaba/GraphScope/releases) 都会对应一个 tag，如 `registry.cn-hongkong.aliyuncs.com/graphscope:0.13.0`

**Comment by hh2712** [comment]
我看到好像graph-learn里面有支持Pytorch的分支，如果直接切那个分支，用graphscope-dev镜像生成，会有兼容性的问题么？

**Comment by sighingnow** [comment]
> 用graphscope-dev镜像生成，会有兼容性的问题么？

As we have discussed in the DingDing group, the `graphscope-rebase` branch in the graph-learn repo should be what you want, but it hasn't been tested with the cases, demos, and tutorials in graphscope.

We already have some effort on that, and hope to included the new implementation with revised test suites in the next release, which will be hopefully delivered early the next month.

**Comment by sighingnow** [comment]
Close as the original issue in this ticket has been addressed. Please feel free to open new issues if you have further questions about graphscope.
